### PR TITLE
Feature/working for families

### LINKED ISSUE #19: Working For Families Family Tax Credit
needs the logic in #14 to work out that they're not eligible for orphans benefit.

### PR DESCRIPTION
**Pull request by dlouise64**
Tests working for:

WFF Tax Credit
WFF In Work Tax Credit
WFF Minimum Family Tax Credit
WFF Parental Tax Credit

closes #19

### CODE REVIEW

File: pytests/test_wff_in_work_tax_credit.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    body = {
+        "applicant": {
+            "isParent": True,
+            "receivesIncomeTestedBenefit": False,
+            "isPrincipalCarer": True,
+        },
+        "income": {
+            "ofApplicantAndSpouse": 0
+        }
```
Comment (Br3nda) [inline pytests/test_wff_in_work_tax_credit.py]: i don't understand how they need an income of zero to get a tax credit.
@piawaugh Do you know what's going on here?

File: pytests/test_wff_ftc_orphans.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        and 20 ≤ applicant.worksWeeklyHours
+        and not applicant.receivesIncomeTestedBenefit
+        and not benefit.isStudentAllowance
+        and applicant.isPrincipalCarer
+        and not benefit.isOrphansBenefit
+            then benefit.isWorkingForFamiliesFamilyTaxCredit is PERMITTED
+"""
+
+
+class TestWorkingForFamiliesFamilyTaxCreditOrphansForbidden(Reasoner):
```
Comment (Br3nda) [inline pytests/test_wff_ftc_orphans.py]: 🎇 !!! Awesome tests

File: pytests/test_wff_min_fam_tax_credit.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    """
+
+    body = {}
+
+    def test_reasoning(self):
+        self.assertTrue(self.is_forbidden)
+        self.assertTrue(self.is_conclusive)
+
+
+class TestWFFMinFamTaxCreditElig(TestKey):
```
Comment (dlouise64) [inline pytests/test_wff_min_fam_tax_credit.py]: @Br3nda Can you please take a look at this test that is failing, should be working, but I'm also getting Forbidden in Raap.

File: pytests/test_wff_family_tax_credit.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        },
+        "benefit": {
+            "isStudentAllowance": False,
+            "isOrphansBenefit": False
+        },
+        "parents": {
+            "areDeceasedMissingOrIncapableThroughDisability": False
+        },
+        "child": {
+            "isDependent": False
```
Comment (Br3nda) [inline pytests/test_wff_family_tax_credit.py]: Just learned we're gonna need to change this to a new atom: `child.isFinanciallyIndependent` , which is different to being dependent. 

A child is financially dependent (but still dependent) if they are working 30 hours a week, getting student allowance, or on their own benefit/other govt assistance

Comment (dlouise64) [reply pytests/test_wff_family_tax_credit.py]: @Br3nda Thanks I will mention this in the meeting today.

**Comment by Br3nda** [comment]
i lost track of this. Were the changes made? not needed... ?

File: pytests/test_wff_family_tax_credit.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    """
+
+    body = {}
+
+    def test_reasoning(self):
+        self.assertTrue(self.is_forbidden)
+        self.assertTrue(self.is_conclusive)
+
+
+class TestWFFTaxCredit(TestWFFTaxCreditKey):
```
[the remark this answers is not in the document]
Comment (Br3nda) [reply pytests/test_wff_family_tax_credit.py]: Needs to give the classes unique name

**Comment by Br3nda** [comment]
I've removed the duplicate default tests

File: pytests/test_child_disability_allowance.py [hunk 6]
```diff
@@ -1,58 +1,55 @@
...
-            # "hasSeriousDisability": True,
-            # "requiresConstantCare": True,
-            # "hasMedicalCertification": True
-        }
-    }
-
-    def test_reasoning(self):
-        self.assertTrue(self.is_conclusive)
-        self.assertTrue(self.is_forbidden)
+# from . import Reasoner
```
Comment (Br3nda) [inline pytests/test_child_disability_allowance.py]: Not sure why this is commented out in a WFF PR

**Comment by Br3nda** [comment]
Still falling :(

**Comment by dlouise64** [comment]
@Br3nda yeah this is a bit of a mess now with the new changes. Also I noticed the rules seem a bit overcomplicated. I think we will need to rewrite the rules as we did with Childcare subsidy. We can look at this again on Monday.

File: pytests/test_wff_in_work_tax_credit.py [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    D. Forbidden is receiving "Parental Tax Credit"
+    """
+
+    body = {
+        "applicant": {
+            "isParent": True,
+            "receivesIncomeTestedBenefit": False,
+            "isPrincipalCarer": True,
+            "relationshipStatus": "single",
+            "weeklyHours": 21
```
Comment (jenofdoom) [inline pytests/test_wff_in_work_tax_credit.py]: Should be worksWeeklyHours ?

File: pytests/test_wff_in_work_tax_credit.py [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+        },
+        "income": {
+            "ofApplicantAndSpouse": 2
```
Comment (jenofdoom) [inline pytests/test_wff_in_work_tax_credit.py]: Should now use the threshold?

File: pytests/test_wff_in_work_tax_credit.py [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+        },
+        "benefit": {
+            "isStudentAllowance": False
```
Comment (jenofdoom) [inline pytests/test_wff_in_work_tax_credit.py]: Surely we shouldn't base the rule on them being eligible for another benefit, as we don't know if they have applied for it yet. If we need to rule this out, we should have an atom to ask if they are currently receiving it.

**Pull request merged**